### PR TITLE
podman-top: use containers/psgo

### PR DESCRIPTION
**Pull request by vrothberg**
Use github.com/containers/psgo instead of execing `ps (1)`.  The psgo
library enables a much more flexible interface with respect to which
data to be printed (e.g., capabilities, seccomp mode, PID, PCPU, etc.)
while the output can be parsed reliably.  The library does not use
ps (1) but parses /proc and /dev instead.  To list the processes of a
given container, psgo will join the mount namespace of the given
container and extract all data from there.

Notice that this commit breaks compatibility with docker-top.

Signed-off-by: Valentin Rothberg <vrothberg@suse.com>

### CODE REVIEW

**Comment by vrothberg** [comment]
@rhatdan @baude PTAL

Note that the manpage of docker-top needs to be updated as well. Once I have your LGTM on the code, I can update the manpage if you wish.

I also want to add extend the bash-completion to cover all supported format descriptors, which requires a change to containers/psgo first.

File: cmd/podman/top.go [hunk 1]
```diff
@@ -18,7 +18,9 @@ var (
 	topDescription = `
    podman top
```
Comment (mheon) [inline cmd/podman/top.go]: Indentation is wierd here - looks like this is using spaces while the new description uses tabs?

**Comment by mheon** [comment]
We should document the allowed format descriptors somewhere in the manpages, but the code LGTM

File: cmd/podman/top.go (hunk 1 above)
Comment (vrothberg) [reply cmd/podman/top.go]: Good catch! I will fix that and make it consistent with other commands (e.g., `podman-pull`).

**Comment by vrothberg** [comment]
Assuming Travis is happy now, I'll commit ontop here to add bash-completion and add the flags to the manpage.

**Comment by vrothberg** [comment]
Updated the PR with latest containers/psgo, manpage changes, bash-completion adjustment to include the format descriptors etc.

**Comment by rhatdan** [comment]
Ok lets get this in.  We know we have some other issues, but we can fix them as psgo evolves.

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**